sdk/opendp/smartnoise/_ast/ast.py

```diff
@@ -254,7 +254,7 @@ def symbol(self, expression):
             symbol
             for name, symbol in syms_b
             if (type(symbol) is TableColumn and symbol.compare.identifier_match(colname, name))
-            or name.lower() == colname.lower()
+            or name == colname
```

Comment:
Line 256 enforces engine-specific comparison rules for `TableColumn`, and was working fine.  Line 257 is supposed to catch cases where the column being compared is not a `TableColumn`, but is instead a computed column like `ROW_NUMBER`.  However, this line was using `.lower()`, and would apply to `TableColumn` and non-`TableColumn` columns, which could result in a `TableColumn` correctly failing to get a match in the test above, and then erroneously matching on the next line.  Switching to use case-sensitive is safer.